### PR TITLE
Kovan - fast sync

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

File: src/Nethermind/Nethermind.Blockchain/Synchronization/SyncConfig.cs [hunk 1]
```diff
@@ -42,5 +42,6 @@ public class SyncConfig : ISyncConfig
         public int BeamSyncPreProcessorTimeout { get; set; } = 15;
         public bool BeamSyncFixMode { get; set; } = false;
         public bool BeamSyncVerifiedMode { get; set; } = false;
+        public long[] TotalDifficultyOverrides { get; set; }
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Blockchain/Synchronization/SyncConfig.cs:45]: can we add it in Chainspec instead?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Blockchain/Synchronization/SyncConfig.cs:45]: We can, the question is do we want to introduce more incompatibilities with parity chainspec? What do you think?

File: src/Nethermind/Nethermind.Runner/configs/kovan.cfg [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/configs/kovan.cfg:1]: do we have config tests for kovan.cfg? there is an extensive config tester in the Nethermind.Runner project

File: src/Nethermind/Nethermind.Serialization.Rlp/ReceiptMessageDecoder.cs [hunk 3]
```diff
@@ -36,18 +37,24 @@ public TxReceipt Decode(RlpStream rlpStream, RlpBehaviors rlpBehaviors = RlpBeha
             }
             
             TxReceipt txReceipt = new TxReceipt();
-            rlpStream.ReadSequenceLength();
+            var lenght = rlpStream.ReadSequenceLength();
             byte[] firstItem = rlpStream.DecodeByteArray();
-            if (firstItem.Length == 1)
+            if (firstItem.Length == 1 && (firstItem[0] == 0 || firstItem[0] == 1))
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Serialization.Rlp/ReceiptMessageDecoder.cs:42]: what is this? why 0 or 1?

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/FastHeadersSyncFeed.cs [hunk 4]
```diff
@@ -112,9 +112,29 @@ public class HeadersSyncFeed : SyncFeed<HeadersSyncBatch?>
...
 
+            _fastSyncExpectedDifficultyAuRaOverride =  GetTotalDifficultyOverrides(); 
+
             Activate();
         }
 
+        // Kovan has some wrong difficulty in early blocks before using proper AuRa difficulty calculation
+        // In order to support that we need to support another pivot
+        private Dictionary<long, ulong>? GetTotalDifficultyOverrides()
+        {
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Synchronization/FastBlocks/FastHeadersSyncFeed.cs:123]: so maybe add it to auraconfig?

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs [hunk 5]
```diff
@@ -176,15 +176,10 @@ private bool TryPrepareReceipts(BlockInfo blockInfo, TxReceipt[] receipts, out T
...
-                    if (receiptsRoot != header.ReceiptsRoot)
-                    {
-                        preparedReceipts = null;
-                    }
-                    else
-                    {
-                        preparedReceipts = receipts;
-                    }
+                    IReleaseSpec releaseSpec = _specProvider.GetSpec(blockInfo.BlockNumber);
+                    preparedReceipts = releaseSpec.ValidateReceipts && new ReceiptTrie(releaseSpec, receipts).RootHash != header.ReceiptsRoot
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs:180]: we cannot simply not validate receipts - very dangerous - can we discuss more

**Review by tkstanczak — CHANGES_REQUESTED** [review]
cannot not validate receipts on sync

File: src/Nethermind/Nethermind.Serialization.Rlp/ReceiptMessageDecoder.cs (hunk 3 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Serialization.Rlp/ReceiptMessageDecoder.cs:42]: In order to support situation when we don't get StatusCode nor StateRoot

File: src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs (hunk 5 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs:180]: We have to (just for first 1 mln blocks)

**Pull request merged**